WordPress/src/main/java/org/wordpress/android/ui/uploads/UploadService.java

```diff
@@ -652,39 +711,44 @@ public void onMediaUploaded(OnMediaUploaded event) {
-                synchronized (sPostsWithPendingMedia) {
-                    Iterator<UploadingPost> iterator = sPostsWithPendingMedia.iterator();
-                    while (iterator.hasNext()) {
-                        UploadingPost uploadingPost = iterator.next();
-                        if (!UploadService.hasPendingOrInProgressMediaUploadsForPost(uploadingPost.postModel)) {
-                            if (uploadingPost.isCancelled) {
-                                // Finished all media uploads for a post upload that was cancelled (probably because
-                                // it was re-opened in the editor) - we can just remove it from the list now
-                                iterator.remove();
-                            } else {
-                                // Fetch latest version of the post, in case it has been modified elsewhere
-                                PostModel latestPost = mPostStore.getPostByLocalPostId(uploadingPost.postModel.getId());
-
-                                // Replace local with remote media in the post content
-                                PostModel updatedPost = updatePostWithCurrentlyCompletedUploads(latestPost);
-                                // also do the same now with failed uploads
-                                updatedPost = updatePostWithCurrentlyFailedUploads(updatedPost);
-                                // finally, save the PostModel
-                                mDispatcher.dispatch(PostActionBuilder.newUpdatePostAction(updatedPost));
-
-                                // TODO Should do some extra validation here
-                                // e.g. what if the post has local media URLs but no pending media uploads?
-                                iterator.remove();
-                                mPostUploadHandler.upload(updatedPost);
-                            }
-                        }
+                cleanUp();
+            }
+            stopServiceIfUploadsComplete();
+        }
+    }
+
+    private void cleanUp() {
+        synchronized (sPostsWithPendingMedia) {
+            Iterator<UploadingPost> iterator = sPostsWithPendingMedia.iterator();
+            while (iterator.hasNext()) {
+                UploadingPost uploadingPost = iterator.next();
+                if (!UploadService.hasPendingOrInProgressMediaUploadsForPost(uploadingPost.postModel)) {
+                    if (uploadingPost.isCancelled) {
+                        // Finished all media uploads for a post upload that was cancelled (probably because
+                        // it was re-opened in the editor) - we can just remove it from the list now
+                        iterator.remove();
+                    } else {
+                        // Fetch latest version of the post, in case it has been modified elsewhere
+                        PostModel latestPost = mPostStore.getPostByLocalPostId(uploadingPost.postModel.getId());
```

Comment:
Do we know that all changes to the post will already have been saved to the store when this is called?